pkg/maps/recorder/ipv6.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package recorder
+
+import (
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/types"
+)
+
+type CaptureWcard6 struct {
+	DestAddr types.IPv6 `align:"daddr"`
+	SrcAddr  types.IPv6 `align:"saddr"`
+	DestPort uint16     `align:"dport"`
+	SrcPort  uint16     `align:"sport"`
+	NextHdr  uint8      `align:"nexthdr"`
+	DestMask uint8      `align:"dmask"`
+	SrcMask  uint8      `align:"smask"`
+	Flags    uint8      `align:"flags"`
+}
+
+type CaptureRule6 CaptureRule
+
+func (k *CaptureWcard6) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+func (k *CaptureWcard6) NewValue() bpf.MapValue { return &CaptureRule6{} }
+
+func (k *CaptureWcard6) DeepCopyMapKey() bpf.MapKey {
+	return &CaptureWcard6{
+		DestAddr: k.DestAddr,
+		SrcAddr:  k.SrcAddr,
+		DestPort: k.DestPort,
+		SrcPort:  k.SrcPort,
+		NextHdr:  k.NextHdr,
+		DestMask: k.DestMask,
+		SrcMask:  k.SrcMask,
+		Flags:    k.Flags,
+	}
+}
+
+func (k *CaptureWcard6) Dump(sb *strings.Builder) {
+	sb.WriteString(fmt.Sprintf("[%s/%d]:%d -> [%s/%d]:%d %s ",
+		k.SrcAddr,
+		int(k.SrcMask),
+		k.SrcPort,
+		k.DestAddr,
+		int(k.DestMask),
+		k.DestPort,
+		dumpNextHdr(int(k.NextHdr))))
```

Comment:
nit:
```suggestion
		u8proto.U8proto(k.NextHdr)
```

Should allow you to get rid of `dumpNextHdr`